cdap-hbase-compat-base/src/main/java/co/cask/cdap/data2/util/hbase/HBaseVersion.java

```diff
@@ -243,13 +243,14 @@ static Version determineVersionFromVersionString(String versionString) throws Pa
     if (versionString.startsWith(HBASE_98_VERSION)) {
       return Version.HBASE_98;
     }
-    VersionNumber ver = VersionNumber.create(versionString);
     if (versionString.startsWith(HBASE_10_VERSION)) {
+      VersionNumber ver = VersionNumber.create(versionString);
```

Comment:
Instead of calling `VersionNumber.create(versionString);` only in certain versions, maybe modify this method to handle all known version patterns?